### PR TITLE
support ci on forks

### LINKED ISSUE #14575: Allow running Protocol CI on forked repositories
There is increase of contribution for ITN code from forked repositories. 

Currently our CI build system is not prepared to handle change from fork repo to mina. There are various problems like:
- many jobs checkout origin branches as reference to some comparison and they will checkout branch from fork repo rather than mina (for example merges cleanly)
- we should not allow to publish dockers to docker hub from forked repositories
-  we need to tag dockers properly not to override main repo based dockers

### PR DESCRIPTION
**Pull request by dkijania**
Currently our CI build system is not prepared to handle change from fork repo to mina. There are various problems like:

many jobs checkout origin branches as reference to some comparison and they will checkout branch from fork repo rather than mina (for example merges cleanly)
we should not allow to publish dockers to docker hub from forked repositories
we need to tag dockers properly not to override main repo based dockers

This PR addresses above problems. It adds yet another remote (mina) which always point to mina protocol repository. If CI is run against forked repository it uses branches from mina remote for comparison instead of origin which point to forked repositories. It also changes behavior when running export env variables script. It does not allow to publish dockers and debian packages stable channels or to dockerhub.

Closes #14575

### CODE REVIEW

File: buildkite/scripts/merges-cleanly.sh [hunk 1]
```diff
@@ -4,21 +4,23 @@ BRANCH=$1
...
 
+
 # Adapted from this stackoverflow answer: https://stackoverflow.com/a/10856937
 # The git merge-tree command shows the content of a 3-way merge without
 # touching the index, which we can then search for conflict markers.
 
 # Tell git where to find ssl certs
 git config --global http.sslCAInfo /etc/ssl/certs/ca-bundle.crt
-# Fetch a fresh copy of the repo
-git fetch origin
```
Comment (SanabriaRusso) [inline buildkite/scripts/merges-cleanly.sh:14]: Why not fetching fork? @dkijania

Comment (SanabriaRusso) [reply buildkite/scripts/merges-cleanly.sh:14]: Also in line #31 below: The stderr produced might be wrong as it does not consider $REMOTE

File: buildkite/scripts/run-snark-transaction-profiler.sh [hunk 2]
```diff
@@ -29,5 +29,5 @@ K=1
 MAX_NUM_UPDATES=4
 MIN_NUM_UPDATES=2   
 
-echo "--- Run Snark Transaction Profiler with parameters: --zkapps --k ${K} --max-num-updates ${MAX_NUM_UPDATES} --min-num-updates ${MIN_NUM_UPDATES}"
-python3 ./scripts/snark_transaction_profiler.py ${K} ${MAX_NUM_UPDATES} ${MIN_NUM_UPDATES}
\ No newline at end of file
+echo "-- Run Snark Transaction Profiler with parameters: --zkapps --k ${K} --max-num-updates ${MAX_NUM_UPDATES} --min-num-updates ${MIN_NUM_UPDATES}"
```
Comment (SanabriaRusso) [inline buildkite/scripts/run-snark-transaction-profiler.sh:32]: Maybe unnecessary change?

File: buildkite/src/Monorepo.dhall [hunk 3]
```diff
@@ -28,6 +28,7 @@ let jobs : List JobSpec.Type =
 let prefixCommands = [
   Cmd.run "git config --global http.sslCAInfo /etc/ssl/certs/ca-bundle.crt", -- Tell git where to find certs for https connections
   Cmd.run "git fetch origin", -- Freshen the cache
```
Comment (SanabriaRusso) [inline buildkite/src/Monorepo.dhall:30]: Would this fetch be unnecessary when executing pipelines for a fork? If not required I would add a condition here. It would save us the burden to download the ~450MB repo.

File: buildkite/scripts/run-snark-transaction-profiler.sh (hunk 2 above)
Comment (dkijania) [reply buildkite/scripts/run-snark-transaction-profiler.sh:32]: yes this change was only to trigger CI

File: buildkite/scripts/merges-cleanly.sh (hunk 1 above)
Comment (dkijania) [reply buildkite/scripts/merges-cleanly.sh:14]: In this job we want to compare branch which is already check out against mina repo so we don't need to refresh fork

Comment (dkijania) [reply buildkite/scripts/merges-cleanly.sh:14]: Stderr is correct as we compare forked branch against mina branch

**Review by SanabriaRusso — APPROVED** [review]

**Comment by mrmr1993** [comment]
!ci-build-me
